kratos/geometries/nurbs_volume_geometry.h

```diff
@@ -885,6 +885,42 @@ class NurbsVolumeGeometry : public Geometry<typename TContainerPointType::value_
     ///@name Operations
     ///@{
 
+/**
+    * @brief Returns local coordinates in return for a point in physical coordinates.
+    *        This function assumes the initial geometry to be a regular grid. Only linear mapping is applied.
+    *
+    * @param rPointGlobalCoordinates the point to which the
+    *        projection has to be found.
+    * @param rProjectedPointLocalCoordinates the location of the
+    *        projection in local coordinates.
+    * @param Tolerance not used.
+    * @return 1
+    */
+    int ProjectionPointGlobalToLocalSpace(
+        const CoordinatesArrayType& rPointGlobalCoordinates,
+        CoordinatesArrayType& rProjectedPointLocalCoordinates,
+        const double Tolerance = std::numeric_limits<double>::epsilon()
+    ) const override
+    {
+        const CoordinatesArrayType& lower_point = this->begin()->GetInitialPosition();
+        const CoordinatesArrayType& upper_point = (this->end()-1)->GetInitialPosition();
+
+        const Vector& knots_u = this->KnotsU();
+        const SizeType n_knots_u = this->NumberOfKnotsU();
+        const Vector& knots_v = this->KnotsV();
+        const SizeType n_knots_v = this->NumberOfKnotsV();
+        const Vector& knots_w = this->KnotsW();
+        const SizeType n_knots_w = this->NumberOfKnotsW();
```

Comment:
The overhead should be these function calls. Since now, we do them for each projection.